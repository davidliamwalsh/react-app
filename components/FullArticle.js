import { Component } from 'react'
import moment from 'moment'

class FullArticle extends Component {
  render () {

    const { article } = this.props

    return <ul className="c-article"> 
      <li className="c-article__title">
        {article.title}
      </li>
      <hr className="c-main__head-line" />
      <li className="c-article__image">
        <img src={`${process.env.RUBY_HOST}${article.image}`} />
      </li>
      <li className="c-article__body">
        {article.body}
      </li>
      <li className="c-article__date">
        Posted {moment(article.createdAt).fromNow()}
      </li>
    </ul>
  }
}

export default FullArticle