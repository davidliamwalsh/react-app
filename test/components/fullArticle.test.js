import FullArticle from '../../components/FullArticle'

describe('<FullArticle />', () => {
  it('it renders correctly', () => {
    const article = {
      slug: '1',
      title: 'Article Title',
      body: 'Article body',
      created_at: new Date(Date.UTC(97, 1, 2, 3, 4, 5))
    }
    const component = shallow(
      <FullArticle article={article} />
    )

    expect(component).toMatchSnapshot()
  })
}) 