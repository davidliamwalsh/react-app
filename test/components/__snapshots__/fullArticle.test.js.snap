// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullArticle /> it renders correctly 1`] = `ShallowWrapper {}`;
